server/src/main/java/org/elasticsearch/index/cache/query/QueryCacheStats.java

```diff
@@ -63,6 +69,14 @@ public void add(QueryCacheStats stats) {
         missCount += stats.missCount;
         cacheCount += stats.cacheCount;
         cacheSize += stats.cacheSize;
+        if (ramBytesUsed < -1 && (ramBytesUsed + stats.ramBytesUsed >= 0)) {
```

Comment:
@jakelandis, I added a clarifying comment there. That negative query cache size value is an error in ES's tracking of that stat. I want to log the first time a query cache size stat goes negative in case that gives any hints about the conditions or code area that are responsible for that. Since the query cache size is likely to stay negative from that point on, I don't want to continually spam the logs with that message and that's what the `&& (ramBytesUsed + stats.ramBytesUsed >= 0)` part of the condition tests.